### ISSUE TITLE
Function length(g) returns incorrect values.

### ISSUE THREAD

**Issue by simonschoelly** [JuliaGraphs/LightGraphs.jl#1145]
If `g` is a graph, then `length(g)` returns the number of entries (zero or not)  in the adjacency matrix of `g`. Unfortunately, the return type of `length(g)` is currently the same as `eltype(g)`. This leads to some unwanted behavior: 

```
julia> g1 = SimpleGraph(100)
{100, 0} undirected simple Int64 graph
julia> length(g1)
10000

julia> g2 = SimpleGraph(Int8(100))
{100, 0} undirected simple Int8 graph

julia> length(g2)
16
```
The reason is, that in the code we do `nv(g) * nv(g)` which overflows for `Int8`. An easy solution would be to convert `nv(g)` to `Int` before multiplying, so that the return type of `length` is always `Int`.

The function can be found in `src/operators.jl`

**Comment by sbromberger** [comment]
More correct would be to use `widen`, I think.

**Comment by simonschoelly** [comment]
Not sure if if `widen` is the correct approach here, `length` returns `Int` for almost all arguments in `Base`. The only exception that I found where for ranges of `BigInt`.

**Comment by simonschoelly** [comment]
This has been resolved by #1155, so I am closing this.
